File: Ejercicios Propuestos 2/Ejercicio 6/Ejercicio 6.jl
println("Ingrese el DÍA de la primera fecha (Escriba 0 para salir):")
dia1 = parse(Int, readline())

# El ciclo se repite MIENTRAS el día 1 no sea cero
while dia1 != 0
    global dia1  # Indicamos que modificaremos dia1 dentro del bucle al final
    
    # Como el día no es 0, terminamos de pedir la primera fecha
    println("Ingrese el MES de la primera fecha:")
    mes1 = parse(Int, readline())
    println("Ingrese el AÑO de la primera fecha:")
    año1 = parse(Int, readline())
    
    # Pedimos la segunda fecha completa
    println("--- Segunda Fecha ---")
    println("Ingrese el DÍA de la segunda fecha:")
    dia2 = parse(Int, readline())
    println("Ingrese el MES de la segunda fecha:")
    mes2 = parse(Int, readline())
    println("Ingrese el AÑO de la segunda fecha:")
    año2 = parse(Int, readline())
    
    println("--------------------------------------------------")
    
    # Comparación de mayor a menor
    if año1 < año2
        println("-> La primera fecha (", dia1, "/", mes1, "/", año1, ") es anterior.")
    else
        if año1 > año2
            println("-> La segunda fecha (", dia2, "/", mes2, "/", año2, ") es anterior.")
        else
            # Si los años son iguales. Comparamos los meses:
            if mes1 < mes2
                println("-> La primera fecha es anterior.")
            else
                if mes1 > mes2
                    println("-> La segunda fecha es anterior.")
                else
                    # Si los años y meses son iguales. Comparamos los días:
                    if dia1 < dia2
                        println("-> La primera fecha es anterior.")
                    else
                        if dia1 > dia2
                            println("-> La segunda fecha es anterior.")
                        else
                            println("-> Ambas fechas son exactamente iguales.")
                        end
                    end
                end
            end
        end
    end
    
    println("--------------------------------------------------")
    # Volvemos a pedir el primer día para ver si el ciclo continúa o se detiene
    println("Ingrese el DÍA de la siguiente primera fecha (Escriba 0 para salir):")
    dia1 = parse(Int, readline())
end

println("Programa finalizado exitosamente.")

File: Ejercicios Propuestos 2/Ejercicio 6/ejercicio6.cpp
#include <iostream>

using namespace std;

int main() {
    int dia1, mes1, anio1;
    int dia2, mes2, anio2;
    
    cout << "Ingrese el DÍA de la primera fecha (Escriba 0 para salir):" << endl;
    cin >> dia1;
    
    // El ciclo se repite mientras el día 1 no sea cero
    while (dia1 != 0) {
        // Como el día no es 0, terminamos de pedir la primera fecha
        cout << "Ingrese el MES de la primera fecha:" << endl;
        cin >> mes1;
        cout << "Ingrese el AÑO de la primera fecha:" << endl;
        cin >> anio1;
        
        // Pedimos la segunda fecha completa
        cout << "--- Segunda Fecha ---" << endl;
        cout << "Ingrese el DÍA de la segunda fecha:" << endl;
        cin >> dia2;
        cout << "Ingrese el MES de la segunda fecha:" << endl;
        cin >> mes2;
        cout << "Ingrese el AÑO de la segunda fecha:" << endl;
        cin >> anio2;
        
        cout << "--------------------------------------------------" << endl;
        
        // Comparación de mayor a menor
        if (anio1 < anio2) {
            cout << "-> La primera fecha (" << dia1 << "/" << mes1 << "/" << anio1 << ") es anterior." << endl;
        } else {
            if (anio1 > anio2) {
                cout << "-> La segunda fecha (" << dia2 << "/" << mes2 << "/" << anio2 << ") es anterior." << endl;
            } else {
                // Si los años son iguales. Comparamos los meses:
                if (mes1 < mes2) {
                    cout << "-> La primera fecha es anterior." << endl;
                } else {
                    if (mes1 > mes2) {
                        cout << "-> La segunda fecha es anterior." << endl;
                    } else {
                        // Si los años y meses son iguales. Comparamos los días:
                        if (dia1 < dia2) {
                            cout << "-> La primera fecha es anterior." << endl;
                        } else {
                            if (dia1 > dia2) {
                                cout << "-> La segunda fecha es anterior." << endl;
                            } else {
                                cout << "-> Ambas fechas son exactamente iguales." << endl;
                            }
                        }
                    }
                }
            }
        }
        
        cout << "--------------------------------------------------" << endl;
        // se pide sino se vuelve a hacer
        cout << "Ingrese el DÍA de la siguiente primera fecha (Escriba 0 para salir):" << endl;
        cin >> dia1;
    }
    
    cout << "Programa finalizado exitosamente." << endl;
    
    return 0;
}